### ISSUE TITLE
Dereferencing a hardcoded address pointer declared as const on RHS creates compile error: "unreachable: ir.cpp:const_ptr_pointee"

### ISSUE THREAD

**Issue by vegecode** [ziglang/zig#1171]
```zig
const myP = @intToPtr(*u16, 0x2300);

pub fn main() void {
  myP.* = myP.* | 10;
}
```

This causes a compiler crash. With --verbose-ir:

```
{
    myP.* = (myP.* | 10);
}
{ // (IR)
Entry_0:
    #1  | (unknown)   | 1 | declref ptr myP
    #2  | (unknown)   | 1 | * #1
    #3  | (unknown)   | 1 | declref myP
    #4  | (unknown)   | 1 | * #3
    #5  | comptime_int| 1 | 10
    #6  | (unknown)   | 1 | #4 | 10
    #7  | void        | - | *#2 = #6
    #8  | void        | 1 | {}
    #9  | (unknown)   | - | @checkStatementIsVoid({})
    #10 | void        | 0 | {}
    #11 | void        | 2 | {}
    #12 | (unknown)   | - | @addImplicitReturnType({})
    #13 | noreturn    | - | return {}
}

Source: @intToPtr(*u16, 8960)
{ // (IR)
Entry_0:
    #1  | type        | 1 | u16
    #2  | (unknown)   | - | &:0:0 u16
    #3  | comptime_int| 1 | 8960
    #4  | (unknown)   | 2 | @intToPtr(#2,8960)
    #5  | (unknown)   | - | @addImplicitReturnType(#4)
    #6  | noreturn    | - | return #4
}
{ // (analyzed)
Entry_0:
    #7  | noreturn    | - | return (*u16)(2300)
}
unreachable: /home/justin/Code/Zig/zig/src/ir.cpp:const_ptr_pointee:166
```

The intended usage is for writing to memory mapped registers.

**Comment by vegecode** [comment]
Dereferencing the const pointer at all on the RHS makes an error:
```
const myP = @intToPtr(*volatile u16, 0x340);
const assert = @import("std").debug.assert;

pub fn main() void {
  assert(@typeOf(myP) == *volatile u16); // good
  assert(@typeOf(myP.*) == u16); // causes error
  var temp = myP.*; // also causes error
}
```

If the myP is declared as `var` instead it compiles fine but the --verbose-ir shows an error:
`bigint.cpp:1633: uint64_t bigint_as_unsigned(const BigInt*): Assertion `!bigint->is_negative' failed.`

**Comment by andrewrk** [comment]
Fixed by @vegecode in 1f08be4d7fdf08bda66312ab4b2e401de378083e
